### PR TITLE
Check that halt_slot >= starting_slot

### PR DESCRIPTION
**Pull request by apfitzge**
#### Problem
Error message isn't clear from the connected slots check if halt_slot is specified incorrectly.

#### Summary of Changes
Add a check that `halt_slot >= starting_slot` before checking connected slots.

Fixes #

### CODE REVIEW

**Comment by apfitzge** [comment]
@HaoranYi @steviez (still in progress but tagging you)

**Comment by apfitzge** [comment]
shred-version subcommand still also needs to be fixed since that specifies `Some(0)` as well.

File: ledger-tool/src/main.rs [hunk 1]
```diff
@@ -1066,7 +1066,13 @@ fn load_bank_forks(
     };
 
     if let Some(halt_slot) = process_options.halt_at_slot {
-        // Check if we have the slot data necessary to replay from starting_slot to >= halt_slot.
+        if halt_slot < starting_slot {
+            eprintln!(
+                "Unable to load bank forks at slot {halt_slot} because it is less than the starting slot {starting_slot}."
```
[the remark this answers is not in the document]
Comment (apfitzge) [reply ledger-tool/src/main.rs]: Done.

**Review by steviez — APPROVED** [review]
LGTM - this should definitely be much more helpful for those not intimately familiar with `ledger-tool

**Comment by HaoranYi** [comment]
Looks good to me too. Thanks!

**Pull request merged**